Initial commit of pipeline

mook-as opened the pull request:
This currently uploads a bunch of tar files, but no docker images yet.

mook-as committed:
Initial commit of pipeline

This currently uploads a bunch of tar files, but no docker images yet.

aarondl commented on cf-suse-stacks.yml:
```diff
@@ -0,0 +1,85 @@
+  source:
+    endpoint: {{s3-endpoint}}
+    bucket: {{s3-bucket}}
+    regexp: opensusefs2-(.*).tar.gz$
```
Isn't this supposed to be named opensusefs42?

mook-as replied:
Err, I don't know? Happy to change, though.

mook-as replied:
Renamed in f2bf75377a39066dda6cdd3aa969b3f457690852

aarondl commented on tasks/make-rootfs/run.sh:
```diff
@@ -0,0 +1,27 @@
+
+SUFFIX="${STACKS_SUFFIX:-}"
+
+src-ci/concourse/tasks/common/start-docker.sh
```
Before I even mention this I realize the insanity of the comment:
Should we not use the concourse docker resource for this? Instead of this script that's caused us issues in the past? And yes, I know that it's the same script. I think that there's benefits in getting later versions from upstream that may solve problems with it before we run into them.

mook-as replied:
No, because [cf-suse-stacks](https://github.com/SUSE/cf-suse-stacks/blob/master/Makefile) runs `docker` all over the place :(

If _that_ gets changed to not do that, then yes.

The script was copied (fresh) from [the upstream job], just with a different name. And the fixed so it's not broken.

[the upstream job]: https://github.com/cloudfoundry/buildpacks-ci/blob/master/scripts/start-docker

aarondl commented on tasks/make-rootfs/task.yml:
```diff
@@ -0,0 +1,18 @@
+  - name: receipt
+outputs:
+  - name: out
+#outputs:
```
Commented things?

mook-as replied:
Oops, deleted in f23ac7dcee4b61b570b84af7c31ba04c24a1a749

aarondl commented on tasks/make-rootfs/run.sh:
```diff
@@ -0,0 +1,27 @@
+
+versioned_receipt_filename="out/opensusefs2${SUFFIX}-$(cat version/number).receipt.txt"
+echo "Rootfs SHASUM: $(sha1sum "$versioned_stack_filename" | awk '{print $1}')" > "$versioned_receipt_filename"
+echo "" >> "$versioned_receipt_filename"
```
Is this literally just to get a newline in the file? `\n` on the end with `-e` doesn't work? Might break other things. Anyway, just felt weird.

mook-as replied:
Yes, copied from [upstream](https://github.com/cloudfoundry/buildpacks-ci/blob/45199e773ca2a7bc9d63592252c87570398e125c/tasks/make-rootfs/run.sh#L21).

aarondl commented on tasks/make-rootfs/run.sh:
```diff
@@ -0,0 +1,27 @@
+versioned_receipt_filename="out/opensusefs2${SUFFIX}-$(cat version/number).receipt.txt"
+echo "Rootfs SHASUM: $(sha1sum "$versioned_stack_filename" | awk '{print $1}')" > "$versioned_receipt_filename"
+echo "" >> "$versioned_receipt_filename"
+cat src/cflinuxfs2/opensusefs2_zypper.out >> "$versioned_receipt_filename"
```
`cp`?

mook-as replied:
Nope, I'm appending here, `cp` will do the wrong thing.

mook-as committed:
Remove useless commented yaml

mook-as committed:
Rename stack to opensuse42

Some names are still hard-coded as opensusefs2 because they
come from the stack repo.

mook-as committed:
Fix path to task definition

It was referring to a directory that didn't exist

aarondl reviewed (approved):
LGTM
